src/os_posix/os_inmemory.c

```diff
@@ -0,0 +1,463 @@
+/*-
```

Comment:
The file seems misplaced to me.  It doesn't appear as though there is much in this file that is directly posix related and this file should be above the OS layer.  Maybe in support/?